### ISSUE TITLE
"cannot find zipfile directory in one of gmock-1.7.0.zip" error

### ISSUE THREAD

**Issue by aalexand** [grpc/grpc#7952]
Using the below command to build grpc:

```
git clone -b v1.0.0 --recursive https://github.com/grpc/grpc.git &&     cd grpc/third_party/protobuf &&     ./autogen.sh && ./configure &&  make -j8 && make install &&     cd ../.. &&     ldconfig &&     make -j8 && make install &&     cd /tmp &&     rm -rf /tmp/grpc
```

Getting this error:

```
Google Mock not present.  Fetching gmock-1.7.0 from the web...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    49  100    49    0     0    202      0 --:--:-- --:--:-- --:--:--   203
[gmock-1.7.0.zip]
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of gmock-1.7.0.zip or
        gmock-1.7.0.zip.zip, and cannot find gmock-1.7.0.zip.ZIP, period.
```

It used to build fine ~1 week ago.

**Comment by pgrosu** [comment]
@aalexand You can get it from here:

http://pkgs.fedoraproject.org/repo/pkgs/gmock/gmock-1.7.0.zip/073b984d8798ea1594f5e44d85b20d66/gmock-1.7.0.zip

**Comment by aalexand** [comment]
Get and put where?

Alternatively, is there a way to avoid grpc wanting this dependency?  I am
not going to run any tests, I am just building and installing grpc itself.

**Comment by pgrosu** [comment]
Just update this line and rerun it:

https://github.com/grpc/grpc/blob/bceec94ea4fc5f0085d81235d8e1c06798dc341a/tools/run_tests/build_artifact_protoc.bat#L35

**Comment by aalexand** [comment]
I'm running on Linux, this looks a Windows file?

**Comment by pgrosu** [comment]
Sorry my mistake, didn't notice at first that you're building protobuf from scratch.  So in your `grpc/third_party/protobuf` directory edit this line in this file:

https://github.com/google/protobuf/blob/master/autogen.sh#L34

**Comment by pgrosu** [comment]
If not just download it directly into the directory.

**Comment by pgrosu** [comment]
You might need to submodule gRPC.

**Comment by pgrosu** [comment]
You're trying to get 1.0.0 but that branch does not exist anymore:

```
git clone -b v1.0.0 --recursive https://github.com/grpc/grpc.git &&     cd grpc/third_party/protobuf
Initialized empty Git repository in /home/pgrosu/grpc-test/grpc/.git/
kremote: Counting objects: 195841, done.                                                                                                    rremote: Compressing objects: 100% (65/65), done.
remote: Total 195841 (delta 32), reused 0 (delta 0), pack-reused 195776
Receiving objects: 100% (195841/195841), 80.90 MiB | 16.78 MiB/s, done.
Resolving deltas: 100% (147387/147387), done.
warning: Remote branch v1.0.0 not found in upstream origin, using HEAD instead
```

**Comment by pgrosu** [comment]
Try to follow these instructions listed under [INSTALL.md](https://github.com/grpc/grpc/blob/master/INSTALL.md#build-from-source):

```
 $ git clone -b $(curl -L http://grpc.io/release) https://github.com/grpc/grpc
 $ cd grpc
 $ git submodule update --init
 $ make
 $ [sudo] make install
```

**Comment by pgrosu** [comment]
There might be something with your setup - or maybe `curl` - as it works for me:

```
$ cd third_party/protobuf/
$ ./autogen.sh
+ autoreconf -f -i -Wall,no-obsolete
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, `build-aux'.
libtoolize: copying file `build-aux/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIR, `m4'.
libtoolize: copying file `m4/libtool.m4'
libtoolize: copying file `m4/ltoptions.m4'
libtoolize: copying file `m4/ltsugar.m4'
libtoolize: copying file `m4/ltversion.m4'
libtoolize: copying file `m4/lt~obsolete.m4'
configure.ac:30: installing `build-aux/config.guess'
configure.ac:30: installing `build-aux/config.sub'
configure.ac:24: installing `build-aux/install-sh'
configure.ac:24: installing `build-aux/missing'
Makefile.am: installing `build-aux/depcomp'
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, `build-aux'.
libtoolize: copying file `build-aux/ltmain.sh'
libtoolize: Consider adding `AC_CONFIG_MACRO_DIR([m4])' to configure.ac and
libtoolize: rerunning libtoolize, to keep the correct libtool macros in-tree.
libtoolize: Consider adding `-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
configure.ac:25: installing `build-aux/config.guess'
configure.ac:25: installing `build-aux/config.sub'
configure.ac:19: installing `build-aux/install-sh'
configure.ac:19: installing `build-aux/missing'
Makefile.am: installing `build-aux/depcomp'
libtoolize: putting auxiliary files in `.'.
libtoolize: copying file `./ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIR, `m4'.
libtoolize: copying file `m4/libtool.m4'
libtoolize: copying file `m4/ltoptions.m4'
libtoolize: copying file `m4/ltsugar.m4'
libtoolize: copying file `m4/ltversion.m4'
libtoolize: copying file `m4/lt~obsolete.m4'
configure.ac:46: installing `./config.guess'
configure.ac:46: installing `./config.sub'
configure.ac:48: installing `./install-sh'
configure.ac:48: installing `./missing'
benchmarks/Makefile.am: installing `./depcomp'
+ rm -rf autom4te.cache config.h.in~
+ exit 0
$
```

**Comment by aalexand** [comment]
Strange.  As far as I can tell,
https://github.com/grpc/grpc/tree/v1.0.0/third_party refers to
https://github.com/google/protobuf/tree/e8ae137c96444ea313485ed1118c5e43b2099cf1,
where
https://github.com/google/protobuf/blob/e8ae137c96444ea313485ed1118c5e43b2099cf1/autogen.sh
refers to https://googlemock.googlecode.com/files/gmock-1.7.0.zip which
seems a non-working URL:

```
curl -o- https://googlemock.googlecode.com/files/gmock-1.7.0.zip There was
no service found for the uri requested.
```

**Comment by pgrosu** [comment]
I believe you're using an old version of `autogen.sh`, since it got recently updated via this commit:

https://github.com/google/protobuf/commit/bba446bbf2ac7b0b9923d4eb07d5acd0665a8cf0

**Comment by aalexand** [comment]
I am using the versions of submodules as referenced by grpc v1.0.0 tag
content.

**Comment by pcj** [comment]
I noticed the same issue and filed https://github.com/bazelbuild/bazel/pull/1688.  Link to the github release are in the diff.

**Comment by bryanayers** [comment]
I'm seeing the same issue with the "v1.0.0" tag, it wasn't happening a couple weeks ago with "release-0_15_1".  I can provide a Dockerfile if it would help.

**Comment by aalexand** [comment]
FWIW, I've worked around the issue by patching the sources with the fix from master as below.  The "curl ... | patch ..." command does the trick.

```
RUN cd /tmp && \
    git clone -b v1.0.0 --recursive https://github.com/grpc/grpc.git && \
    cd grpc/third_party/protobuf && \
    curl -o- https://github.com/google/protobuf/commit/bba446bbf2ac7b0b9923d4eb07d5acd0665a8cf0.diff | patch -p1 && \
    ./autogen.sh && ./configure &&  make -j8 && make install && \
    cd ../.. && \
    ldconfig && \
    make -j8 && make install && \
    cd /tmp && \
    rm -rf /tmp/grpc
```

**Comment by pgrosu** [comment]
@aalexand Happy to hear you got it working!  That's usually my default process of how I figure out how the APIs/tools are put together - it has always led me to discover some cool things.  Welcome to my world :)

**Comment by Mityuha** [comment]
I had the same problem, and I've run worked around command by @aalexand. And now I have an output 

> --grpc_out--grpc_out: Unimplemented GenerateAll() method.
> --grpc_out: --grpc_out: Unimplemented GenerateAll() method. 
> : Unimplemented GenerateAll() method. 
> Unimplemented GenerateAll() method.

What am I doing wrong?

**Comment by Mityuha** [comment]
Ok. Tell me, please how can I at least install grpc (c++) on Ubuntu 16.04. I've tried to install step by step from  [https://github.com/grpc/grpc/blob/master/INSTALL.md#build-from-source](url), and I've googled all day, but there is no result.

**Comment by apeyser** [comment]
Hmm, could someone fix the last stable 3.0 release so you can build without patching?

**Comment by ky0ncheng** [comment]
[gmock-1.7.0.zip]
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of gmock-1.7.0.zip or
        gmock-1.7.0.zip.zip, and cannot find gmock-1.7.0.zip.ZIP, period.
➜  protobuf-3.0.0 ls
appveyor.bat      generate_descriptor_proto.sh  protobuf.bzl
appveyor.yml      gmock-1.7.0.zip               protobuf-lite.pc.in
autogen.sh        gmock.BUILD                   protobuf.pc.in
